include/zephyr/dt-bindings/gpio/nxp-imx-gpio.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2022 Vestas Wind Systems A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NXP_IMX_GPIO_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NXP_IMX_GPIO_H_
+
+/**
+ * @name GPIO pull strength flags
+ *
+ * The pull strength flags are a Zephyr specific extension of the standard GPIO
+ * flags specified by the Linux GPIO binding. Only applicable for NXP IMX
+ * SoCs.
+ *
+ * The interface supports two different pull strengths:
+ * `WEAK` - The lowest pull strength supported by the HW
+ * `STRONG` - The highest pull strength supported by the HW
+ *
+ * @{
+ */
+/** @cond INTERNAL_HIDDEN */
+#define IMXRT10XX_GPIO_PULL_STRENGTH_POS 9
+#define IMXRT10XX_GPIO_PULL_STRENGTH_MASK (0x1U << IMXRT10XX_GPIO_PULL_STRENGTH_POS)
+/** @endcond */
+
+/** pull up/down strengths (only applies to CONFIG_SOC_SERIES_IMXRT10XX) */
+#define IMXRT10XX_GPIO_PULL_WEAK (0x0U << IMXRT10XX_GPIO_PULL_STRENGTH_POS)
+#define IMXRT10XX_GPIO_PULL_STRONG (0x1U << IMXRT10XX_GPIO_PULL_STRENGTH_POS)
```

Comment:
Could we have just one of these flags (e.g. strong)? That would mean only configurations needing the strong pull-up would need to set it.